src/bitcoin-cli.cpp

```diff
@@ -220,6 +222,46 @@ class BaseRequestHandler
     virtual UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) = 0;
     virtual UniValue ProcessReply(const UniValue &batch_in) = 0;
 };
+/** Get new address */
+class GetNewAddressRequestHandler: public BaseRequestHandler
+{
+public:
+    const int ID_GETNEWADDRESS = 0;
+
+    UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& /* args */) override
+    {
+        const UniValue& request = JSONRPCRequestObj("getnewaddress", NullUniValue, ID_GETNEWADDRESS);
+        return request;
+    }
+
+    UniValue ProcessReply(const UniValue &result) override
+    {
+        return JSONRPCReplyObj(result, NullUniValue, 1);
+    }
+};
+/** Generate blocks */
+class GenerateRequestHandler: public BaseRequestHandler
+{
+public:
+    const int ID_GENERATETOADDRESS = 0;
+
+    UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) override
+    {
+        addr = args.at(1);
+        UniValue paramsObj = RPCConvertValues("generatetoaddress", args);
+        return JSONRPCRequestObj("generatetoaddress", paramsObj, ID_GENERATETOADDRESS);
+    }
+
+    UniValue ProcessReply(const UniValue &result) override
+    {
+        UniValue res(UniValue::VOBJ);
+        res.pushKV("address", addr);
+        res.pushKV("blocks", result.get_obj()["result"]);
+        return JSONRPCReplyObj(res, NullUniValue, 1);
+    }
+protected:
+    std::string addr;
+};
```

Comment:
I think you don't need to do any of these 40 lines of setup as you're not batching. You can use the existing `DefaultRequestHandler` and put the 2 `pushKV` lines in your -generate code section below.  See 06a1bf6c1e1de3e219d2f8eebe187c1ca2962e9d for a semi-related example.

Reply:
Many thanks. A good idea. However, my code is based on a pretty "old" version that differs greatly from https://github.com/bitcoin/bitcoin/commit/06a1bf6c1e1de3e219d2f8eebe187c1ca2962e9d 

Maybe before changing all that I should rebase this code and try to re-implement it again.

Reply:
ISTM the only (or main) change between the example in 06a1bf6c1e1de3e219d2f8eebe187c1ca2962e9d and your code is that you would use CallRPC instead of ConnectAndCallRPC (with the same arguments, just renaming).

Reply:
Ok, then let's see how far this goes.

Reply:
E.g.
```cpp
CallRPC(rh.get(), "getnewaddress", args);
CallRPC(rh.get(), "generatetoaddress", args);
```
etc.

Reply:
Sadly, it wouldn't make much sense to change it, because *CallRPC** isn't exactly the same as **ConnectAndCallRPC**. In my code there is an "default" [CallRPC](https://github.com/brakmic/bitcoin/blob/resurrect-generate-method/src/bitcoin-cli.cpp#L568), which collides with my special "-generate" command that needs of two such calls. So, either I modify this implicit RPCCall thus making it even more complex, or I leave it as it is. I don't remember exactly, because it's been three months since I've worked on this code, but I wrote all that "ceremonial" stuff (extra request handlers etc.) because I didn't want to introduce even more *if-then-else* to find my way around the already complex structure there. But now, as the *ConnectAndCallRPC* has actually moved away this "default" CallRPC I could work on that basis. Hence my question, should I rebase the code.